src/AcceptanceTesting/SpecByExample/SFA.DAS.Payments.AcceptanceTests/Assertions/TransfersAssertions.cs

```diff
@@ -16,8 +15,16 @@ public static void ValidateTransfers(TransfersBreakdown breakdown, List<Transfer
             {
                 foreach (var periodValue in line.TransferAmounts)
                 {
-                    var currentResultAmount = transferResults.FirstOrDefault(x => PeriodNameHelper.GetStringDateFromLongPeriod(x.CollectionPeriodName) == periodValue.PeriodName) != null
-                        ? transferResults.FirstOrDefault(x => PeriodNameHelper.GetStringDateFromLongPeriod(x.CollectionPeriodName) == periodValue.PeriodName).Amount : 0;
+                    var currentResultAmount = transferResults.FirstOrDefault(
+                        x => PeriodNameHelper.GetStringDateFromLongPeriod(x.CollectionPeriodName) == periodValue.PeriodName
+                        && x.SendingAccountId == breakdown.SendingEmployerAccountId
+                        && x.ReceivingAccountId == line.ReceivingEmployerAccountId
+                    ) != null
+                    ? transferResults.FirstOrDefault(
+                            x => PeriodNameHelper.GetStringDateFromLongPeriod(x.CollectionPeriodName) == periodValue.PeriodName
+                            && x.SendingAccountId == breakdown.SendingEmployerAccountId
+                            && x.ReceivingAccountId == line.ReceivingEmployerAccountId
+                    ).Amount : 0;
```

Comment:
would it be better to capture the output of `transferResults.FirstOrDefault` as a variable to then perform the null check on and resultant assignment to `currentResultAmount`?

Reply:
yes it would, I'll change it